diff --git a/sl-5.02/sl.c b/sl-5.02/sl.c
index d8f2d75..72fae69 100644
--- a/sl-5.02/sl.c
+++ b/sl-5.02/sl.c
@@ -36,6 +36,10 @@
 /* sl version 1.00 : SL runs vomitting out smoke.                            */
 /*                                              by Toyoda Masashi 1992/12/11 */
 
+#ifdef EMSCRIPTEN
+#include <emscripten/emscripten.h>
+#endif
+
 #include <curses.h>
 #include <signal.h>
 #include <unistd.h>
@@ -78,11 +82,44 @@ void option(char *str)
     }
 }
 
+#ifdef EMSCRIPTEN
+
+EMSCRIPTEN_KEEPALIVE int get_cols() {
+    return COLS;
+}
+
+EMSCRIPTEN_KEEPALIVE int get_lines() {
+    return LINES;
+}
+
+EMSCRIPTEN_KEEPALIVE int update(int x) {
+    if (LOGO == 1) {
+        if (add_sl(x) == ERR) {
+            return ERR;
+        }
+    } else if (C51 == 1) {
+        if (add_C51(x) == ERR) {
+            return ERR;
+        }
+    } else {
+        if (add_D51(x) == ERR) {
+            return ERR;
+        }
+    }
+    getch();
+    refresh();
+    return OK;
+}
+
+EMSCRIPTEN_KEEPALIVE void stop() {
+    mvcur(0, COLS - 1, LINES - 1, 0);
+    endwin();
+}
+#endif
+
 int main(int argc, char *argv[])
 {
-    int x, i;
-
-    for (i = 1; i < argc; ++i) {
+    for (int i = 1; i < argc; ++i) {
         if (*argv[i] == '-') {
             option(argv[i] + 1);
         }
@@ -95,7 +132,8 @@ int main(int argc, char *argv[])
     leaveok(stdscr, TRUE);
     scrollok(stdscr, FALSE);
 
-    for (x = COLS - 1; ; --x) {
+#ifndef EMSCRIPTEN
+    for (int x = COLS - 1; ; --x) {
         if (LOGO == 1) {
             if (add_sl(x) == ERR) break;
         }
@@ -111,6 +149,8 @@ int main(int argc, char *argv[])
     }
     mvcur(0, COLS - 1, LINES - 1, 0);
     endwin();
+#endif
+    return 0;
 }
 
 
